File: app/src/main/java/com/bsas/androiddevs/manejoerrores/manager/MovieManager.java
package com.bsas.androiddevs.manejoerrores.manager;

import android.content.Context;

import com.bsas.androiddevs.manejoerrores.common.exception.DbException;
import com.bsas.androiddevs.manejoerrores.manager.listener.GetMoviesObserver;
import com.bsas.androiddevs.manejoerrores.repository.MovieRepository;

public class MovieManager {

    private MovieRepository movieRepository;

    public MovieManager(Context context) {
        this.movieRepository = new MovieRepository(context);
    }

    public void getMovies(GetMoviesObserver observer) throws DbException {
        this.movieRepository.getMovies(observer);
    }


}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/manager/MovieManager.kt
package com.bsas.androiddevs.manejoerrores.manager

import com.bsas.androiddevs.manejoerrores.common.Movie
import com.bsas.androiddevs.manejoerrores.repository.MovieRepository

class MovieManager {

    private val movieRepository: MovieRepository = MovieRepository()

    suspend fun getMovies(): List<Movie> {
        return movieRepository.getMovies()
//        throw UIErrorException(R.string.error, null)
//        throw UIAlertException(R.string.alert, null)
    }

}